kratos/containers/generic_variables_list.cpp

```diff
@@ -0,0 +1,57 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//                   Jordi Cotela
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "containers/generic_variables_list.h"
+#include "includes/kratos_components.h"
+
+namespace Kratos
+{
+    void GenericVariablesList::AddVariable(const std::string& rVarName)
+    {
+        //ugly but needed: here we need to get the component type by checking the Kratos components
+        if(KratosComponents<Variable<bool>>::Has(rVarName))
+            mVariables.push_back(KratosComponents<Variable<bool>>::Get(rVarName));
+        else if(KratosComponents<Variable<int>>::Has(rVarName))
+            mVariables.push_back(KratosComponents<Variable<int>>::Get(rVarName));
+        else if(KratosComponents<Variable<unsigned int>>::Has(rVarName))
+            mVariables.push_back(KratosComponents<Variable<unsigned int>>::Get(rVarName));
+        else if(KratosComponents<Variable<double>>::Has(rVarName))
+            mVariables.push_back(KratosComponents<Variable<double>>::Get(rVarName));
+        else if(KratosComponents<Variable<array_1d<double,3>>>::Has(rVarName))
```

Comment:
You should add remaining arrays types, array_1d<double,4>, array_1d<double,6>, array_1d<double,9>

Reply:
the problem with adding other types is that one needs to explicitly define an operator for each type IN EVERY CUSTOM VISITOR.

that is a BIG problem...